src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc

```diff
@@ -0,0 +1,167 @@
+        process = subprocess.Popen(['ros2', 'bag', 'info', f'{bagpath}'], stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        return output.decode('utf-8')
+
+    data = pd.concat([
+        lks.evo.series('/beluga_amcl/pose', 'ape', normalization='long'),
+        lks.evo.series('/nav2_amcl/pose', 'ape', normalization='long')
+    ])
+
+    data = data[[
+        'case.name',
+        'variation.parameters.dataset',
+        'variation.parameters.laser_model',
+        'trajectory.name',
+        'variation.parameters.basedir',
+        'metric.series.value'
+    ]]
+
+    data = data.rename(columns = {
+        'case.name': 'case_name',
+        'variation.parameters.dataset': 'bagfile',
+        'variation.parameters.laser_model': 'laser_model',
+        'trajectory.name': 'implementation',
+        'variation.parameters.basedir': 'basedir',
+        'metric.series.value': 'ape',
+    })
+
+    ape = data.groupby([
+        'case_name',
+        'bagfile',
+        'implementation',
+        'laser_model',
+    ])['ape'].agg(['median', 'mean', 'std', 'max']).round(3)
+
+    bagfile_location_data = data.groupby([
+        'case_name',
+        'bagfile',
+    ])['basedir'].agg(['first']).rename(columns={'first': 'basedir'})
+
+    from ament_index_python import get_package_share_directory
+
+    this_pkg_dir = get_package_share_directory('beluga_vs_nav2_multi_dataset')
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_table_header.txt', 'r') as f:
+        TABLE_HEADER_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_table_row.txt', 'r') as f:
+        ROW_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_dataset_chapter.txt', 'r') as f:
+        DATASET_CHAPTER_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_bagfile_section.txt', 'r') as f:
+        BAGFILE_SECTION_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_resource_table_header.txt', 'r') as f:
+        RESOURCE_TABLE_HEADER_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_resource_table_row.txt', 'r') as f:
+        RESOURCE_TABLE_ROW_TEMPLATE = f.read()
+
+    output = []
+
+    def equalize_digits(value, digits=3):
+        return f'{value:.{digits}f}' if value is not str else value
+
+    desired_section_order = [
+        'Magazino Datasets',
+        'Openloris Office',
+        'TorWIC Mapping',
+        'TorWIC SLAM',
+        'Willow Garage',
+        'Diff Drive Sim 24hs',
+        'Omni Drive Sim 24hs',
+    ]
+    datasets_in_data = ape.index.get_level_values(0).unique()
+    missing_datasets = [x for x in datasets_in_data if x not in desired_section_order]
+    desired_section_order = desired_section_order + missing_datasets
+
+    for case_name in desired_section_order:
```

Comment:
@glpuga meta: I wonder if `.. datatemplate:import-module:: lambkin.data` wouldn't be a slightly more sane approach to templating (see https://sphinxcontribdatatemplates.readthedocs.io/en/latest).

Reply:
I'm not sure if this is a feature idea or an alternative implementation. 

In any case, as a user, I'd like LAMBKIN to do **less**, not more, in terms of report generation. 

Ideally it should generate data in easy-to-import-elsewhere formats and leave it at that, and probably one easy to  browse format were I can see the results quickly. 

The usual workflow for the engineer will be I run stuff, and I check results, rerun stuff, check results again, rinse and repeat. In this workflow, any time spent getting the pdf file to work around issues in the data is time wasted, and this is probably 95% of the potential users.

If I need to report elsewhere from time to time, most likely the format will not be anything LAMBKIN can produce : web, ieee templated paper for conference, Google docs, email. Most of the time it will also be a one-off, or infrequent enought that I'll not be willing to spend time cohercing lambkin into producing that format directly.

The dream is a csv I can import in a spreadsheet or reprocess with a trashy python script to convert into a latex table that runs as part of the paper make+latex build pipeline.

Reply:
> I'm not sure if this is a feature idea or an alternative implementation.

Alternative implementation. The current full programmatic approach (as opposed to an interpolated template) is very hard to follow.

> Ideally it should generate data in easy-to-import-elsewhere formats and leave it at that, and probably one easy to browse format were I can see the results quickly.

> The usual workflow for the engineer will be I run stuff, and I check results, rerun stuff, check results again, rinse and repeat. In this workflow, any time spent getting the pdf file to work around issues in the data is time wasted, and this is probably 95% of the potential users.

> If I need to report elsewhere from time to time, most likely the format will not be anything LAMBKIN can produce : web, ieee templated paper for conference, Google docs, email. Most of the time it will also be a one-off, or infrequent enought that I'll not be willing to spend time cohercing lambkin into producing that format directly.

> The dream is a csv I can import in a spreadsheet or reprocess with a trashy python script to convert into a latex table that runs as part of the paper make+latex build pipeline.

Fair point, and you do have that already. Automatic report generation is meant to be a convenience, nevermind the current UX. You can use it but you don't have to use it. You can use [`lambkin.data.access`](https://github.com/Ekumen-OS/lambkin/blob/main/src/core/lambkin-shepherd/src/lambkin/shepherd/data/access.py) APIs on arbitrary benchmark output too, and if you want to use another tool (like a spreadsheet), you can also export the dataframes (once `lambkin` has taken care of aggregating and indexing all the data). It's not well documented though, so there's that.